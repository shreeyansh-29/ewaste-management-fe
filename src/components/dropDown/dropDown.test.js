/* eslint-disable no-undef */
import React from "react";
import Dropdown from "./dropdown";
import Enzyme, {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";

Enzyme.configure({adapter: new Adapter()});

describe("test dropDown Component", () => {
  const props = {
    onChange: jest.fn(),
    data: [{label: "Customer"}, {label: "Collector"}, {label: "Vendor"}],
    value: "",
  };
  const event = {
    preventDefault() {},
    target: {value: "Customer"},
  };

  const name = "Select your Role";
  it("should render dropdown", () => {
    const wrapper = shallow(<Dropdown {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should test onChange", () => {
    const wrapper = shallow(<Dropdown {...props}>{name}</Dropdown>);
    wrapper.find(".dropdown1").simulate("change", event);
    expect(props.onChange).toBeCalledWith("Customer");
  });
});
