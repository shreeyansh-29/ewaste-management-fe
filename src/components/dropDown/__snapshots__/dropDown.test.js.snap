// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dropDown Component should render dropdown 1`] = `
<div>
  <select
    className="dropdown"
    onChange={[Function]}
  >
    <option>
      Select your Role
    </option>
    <option
      key="0"
      value="Customer"
    >
      Customer
    </option>
    <option
      key="1"
      value="Collector"
    >
      Collector
    </option>
    <option
      key="2"
      value="Vendor"
    >
      Vendor
    </option>
  </select>
</div>
`;
